src/kagglesdk/kaggle_http_client.py

```diff
@@ -129,50 +88,13 @@ def call(
         return response
 
     def _prepare_request(self, service_name: str, request_name: str, request: KaggleObject):
-        request_url = self._get_request_url(request)
-        method = request.method()
-        data = ""
-        if method == "GET":
-            data = request.__class__.to_dict(request, ignore_defaults=False)
-            if request.endpoint_path():
-                words = find_words(request.endpoint_path())
-                list(map(data.pop, [to_lower_camel_case(w) for w in words]))
-                if len(data) == 0:
-                    data = None
-            if data:
-                request_url = f"{request_url}?{urllib.parse.urlencode(clean_data(data))}"
-            data = ""
-            self._session.headers.update(
-                {
-                    "Accept": "application/json",
-                    "Content-Type": "text/plain",
-                }
-            )
-        elif method == "POST":
-            data = request.to_field_map(request, ignore_defaults=True)
-            if isinstance(data, dict):
-                fields = request.body_fields()
-                if fields is not None:
-                    if fields != "*":
-                        data = data[fields]
-                data = clean_data(data)
-                if self.requires_form(request):
-                    data, content_type = self.make_form(data)
-                else:
-                    content_type = "application/json"
-                    data = json.dumps(data)
-                self._session.headers.update(
-                    {
-                        "Accept": "application/json",
-                        "Content-Type": content_type,
-                    }
-                )
+        request_url = self._get_request_url(service_name, request_name)
         http_request = requests.Request(
-            method=method,
+            method="POST",
             url=request_url,
-            data=data,
+            json=request.__class__.to_dict(request),
             headers=self._session.headers,
-            # cookies=self._get_xsrf_cookies(),
+            cookies=self._get_xsrf_cookies(),
```

Comment:
We don't need this for ApiV1 endpoints as they don't enforce XSRF checks. I had added it for the experimental admin client to be able to call internal endpoints. IIRC, it fetches the home page to get the XSRF cookie so requries an additional HTTP call per RPC so it is better to remove it for ApiV1